examples/test-exec-doc.yml

```diff
@@ -0,0 +1,153 @@
+# Name of the workflow
+name: Run Tests on Push to Main
+
+# Set permissions required for the workflow
+permissions:
+  contents: write           # Allows writing repository contents
+  pull-requests: write      # Allows creating and modifying pull requests
+  id-token: write           # Allows generating tokens for authentication
+  issues: write             # Allows creating and modifying issues
+
+# Define the events that trigger the workflow
+on:
+  push:
+    paths:
+      - '**/*.md'           # Trigger on push to any Markdown file
+  pull_request:
+    paths:
+      - '**/*.md'           # Trigger on pull requests affecting Markdown files
+  workflow_dispatch:        # Allows manual triggering of the workflow
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest  # Run the job on the latest Ubuntu runner
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0      # Fetch all history for all branches and tags
```

Comment:
Do we need to add the fetch-depth? Is this actually needed?

Reply:
good point - based on GH copilot it isn't so have removed it. will test and see if it breaks at any point while executing